serpentTools/objects/readers.py

```diff
@@ -26,14 +25,35 @@ def __str__(self):
         return '<{} reading {}>'.format(self.__class__.__name__, self.filePath)
 
     def read(self):
+        """The main method for reading that not only parses data, but also
+        runs pre and post checks.
+        """
+        self._precheck()
+        self._read()
+        self._postcheck()
+
+    def _read(self):
         """Read the file and store the data.
 
         :warning:
 
             This read function has not been implemented yet
 
         """
-        raise NotImplementedError
+        pass
```

Comment:
I think this method should raise `NotImplentedError`, ensuring that readers that are not built are not used

Reply:
agreed!